controllers/dspipeline_params.go

```diff
@@ -124,7 +124,7 @@ func (p *DSPAParams) SetupDBParams(ctx context.Context, dsp *dspa.DataSciencePip
 		if p.MariaDB == nil {
 			p.MariaDB = &dspa.MariaDB{
 				Deploy:    true,
-				Image:     config.MariaDBImage,
+				Image:     config.GetStringConfigWithDefault(config.MariaDBImagePath, config.DefaultImageValue),
```

Comment:
@HumairAK the only thing I didn't understand about this PR is this `GetStringConfigWithDefault`. It looks like `DefaultImageValue` is a const set to `MustSetInConfig`, so there's not really any useful fallback value ever returned from `GetStringConfigWithDefault`. What is your intent there? Is it to make sure image pulls fail with `cannot pull docker/MustSetInConfig` ? Is that even what would happen?

Perhaps a comment on `GetStringConfigWithDefault` would help.

Reply:
@gregsheremeta you're right the default here will never be used, it was mostly to just stay consistent with the rest of the pattern and add additional future proof protection to the logic. 

as you already know, this would never happen because we enforce the[ requirement of the Image config](https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/main.go#L80-L84). 

however to me this seemed a bit too decoupled from the rest of the params logic (relying on configs/env vars), so it made sense to have this additional check. Otherwise we are relying on the config file or env variables to be set correctly, are not empty, etc.

Having said that, I'm not against just setting the image directly without using `GetStringConfigWithDefault`.

Reply:
> which pattern do you mean? 

Maybe "rest of the code" would have been more appropriate, essentially the checks we perform on other parameters.  

> what's an example of a future change that this is protecting against? 

* config is made optional/removed
* env vars / configs are empty (accidentally or by design) 

> that makes sense, but I think you could also just panic about it with a direct message instead of causing an image pull error in a roundabout way

We would never get a pull error since this this should never happen any way. 

Maybe a better way to go about this is to just put in a test that checks for all these things, since we're already accounting for it in config logic, I don't mind either way ¯\_(ツ)_/¯